openfisca_core/variables.py

```diff
@@ -162,13 +162,11 @@ def __init__(self, baseline_variable = None):
         self.value_type = self.set(attr, 'value_type', required = True, allowed_values = VALUE_TYPES.keys())
         self.dtype = VALUE_TYPES[self.value_type]['dtype']
         self.json_type = VALUE_TYPES[self.value_type]['json_type']
-        if self.value_type == Enum:
-            self.possible_values = self.set(attr, 'possible_values', required = True, setter = self.set_possible_values)
         if self.value_type == str:
             self.max_length = self.set(attr, 'max_length', allowed_type = int)
-            if self.max_length:
-                self.dtype = '|S{}'.format(self.max_length)
```

Comment:
> @fpagnoux For a value exceeding `max_length`, this line reduced the string to `max_length`. If it was done on purpose, let me know as this behaviour is removed here.

Reply:
I believe the point was to make sure a given variable would be limited to _n_ characters, to reduce the memory usage of the variable(typically, a zip code).

Whether this was a good idea or not can be discussed, but removing this along the way in a bug fix PR is problematic IMO. After this PR, the `max_length` variable attribute is totally useless, yet it hasn't been deprecated, and is still documented. This is not a stable situation.

Reply:
We're reversing this, as it turns out to result in a regression undetected by existing unit tests.

As I recall, @sandcha and I were pairing at the time and we had a miscommunication, I assumed truncation to max_length would still be happening elsewhere as a result of line 168, I had not understood we were eliminating the behaviour.

The truncation behaviour, however, was (still is) untested and implicit. Fixing this properly entails at the very least documenting the truncation behaviour.

@sandcha suggests emitting a truncation warning at execution; I think that does not achieve the purpose of avoiding nasty surprises, because whoever is coding an input variable won't exert control on the actual inputs specified, and whoever actually specifies an overlong input is unlikely to see debug logs from the computation engine.

My preferred solution would be to rename max_length to something like truncate_to_length, to make the behaviour explicit; and a unit test for the truncation behaviour would be ideal.

Reply:
Well, I'm just in favour of giving the information to the user somewhere like in a unit test explaining the expected behaviour. A PR is coming on that 🙂

Reply:
> The truncation behaviour, however, was (still is) untested and implicit. Fixing this properly entails at the very least documenting the truncation behaviour.

👍 

> My preferred solution would be to rename max_length to something like truncate_to_length

👎 Attribute names of a model are preferably _nouns_. `variable.truncate_to_length` sounds like a `function`, because `truncate` is a `verb`.